### ISSUE TITLE
Duplicate symbol 'OBJC_IVAR$_SRIOConsumer._unmaskBytes'...

### ISSUE THREAD

**Issue by zljkevin** [aws-amplify/aws-sdk-ios#4618]
There is a conflict between AWSTranscribeStreaming and SocketRocket；
My podfile is as follows:
```
    pod 'SocketRocket'
    pod 'AWSTranscribeStreaming', '~> 2.30.4'

```
**Logs**

duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._unmaskBytes'...

<details>
<summary>Log Messages</summary>

```
duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._unmaskBytes' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._scanner' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_CLASS_$_SRIOConsumer' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_METACLASS_$_SRIOConsumer' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._handler' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._readToCurrentFrame' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumer._bytesNeeded' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumer.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumerPool._bufferedConsumers' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumerPool.o)
duplicate symbol '_OBJC_CLASS_$_SRIOConsumerPool' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumerPool.o)
duplicate symbol '_OBJC_METACLASS_$_SRIOConsumerPool' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumerPool.o)
duplicate symbol '_OBJC_IVAR_$_SRIOConsumerPool._poolSize' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRIOConsumerPool.o)
duplicate symbol '_SRHTTPResponseErrorKey' in:
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/AWSTranscribeStreaming/libAWSTranscribeStreaming.a(AWSSRWebSocket.o)
    /Users/kuone-ios/Library/Developer/Xcode/DerivedData/ListenNTalk-ewljgkweyutrnffnoomtppmxqbep/Build/Products/Debug-iphoneos/SocketRocket/libSocketRocket.a(SRWebSocket.o)
ld: 12 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
</details>

**Comment by harsh62** [comment]
@zljkevin Apologies for the delayed response on the issue. Were you able to get to a resolution?

**Comment by zljkevin** [comment]
> @zljkevin Apologies for the delayed response on the issue. Were you able to get to a resolution?

I can only modify the AWS source code and rename conflicting names.

**Comment by harsh62** [comment]
Would you be able to share more details about the conflicting names?

**Comment by zljkevin** [comment]
renamed items in class AWSSRWebSocket:
SRHTTPResponseErrorKey → AWSSRHTTPResponseErrorKey 
SRIOConsumer → AWSSRIOConsumer  
SRIOConsumerPool → AWSSRIOConsumerPool
SRHTTPResponseErrorKey → AWSSRHTTPResponseErrorKey 
@implementation NSURL (SRWebSocket) → @implementation NSURL (AWSSRWebSocket)
@implementation NSRunLoop (SRWebSocket) → @implementation NSRunLoop (AWSSRWebSocket)
all SRWebSocket → AWSSRWebSocket

**Comment by ruisebas** [comment]
Released in [2.33.5](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.33.5)
